src/bgg_companion_game_filter.py

```diff
@@ -15,20 +15,33 @@ def filter_games(self) -> list[BoardGame]:
         return filtered_board_games
 
     def game_matches_filter(self, game: BoardGame) -> bool:
+        return (
+                self.__game_type(game=game, gamefilter=self.game_filter)
+                and self.__player_range_type(game=game)
+                and self.__cooperative_type(game=game)
+        )
+
+    def __player_range_type(self, game: BoardGame) -> bool:
         if self.game_filter.playerrangetype == "normal":
             return (
-                self.__game_type(game=game, gamefilter=self.game_filter)
-                and self.__minplayers_in_range(game=game, gamefilter=self.game_filter)
+                self.__minplayers_in_range(game=game, gamefilter=self.game_filter)
                 and self.__maxplayers_in_range(game=game, gamefilter=self.game_filter)
             )
         elif self.game_filter.playerrangetype == "exact":
             return (
-                self.__game_type(game=game, gamefilter=self.game_filter)
-                and self.__minplayers_exact(game=game, gamefilter=self.game_filter)
+                self.__minplayers_exact(game=game, gamefilter=self.game_filter)
                 and self.__maxplayers_exact(game=game, gamefilter=self.game_filter)
             )
         else:
-            return self.__game_type(game=game, gamefilter=self.game_filter)
+            return True
+
+    def __cooperative_type(self, game: BoardGame) -> bool:
+        if self.game_filter.cooperative == True:
```

Comment:
IDE complains about this line thinking the expression can be simplified to 
`if self.game_filter.cooperative`
but that actually breaks tests.  This is probably because there's a huge difference between the 2.  

What I'm doing is specifically setting a conditional for if cooperative is set to true.  
What the IDE wants is seeing if self.game_filter.cooperative exists within the Game Filter. 

I'm just wondering if this is a code sniff, or simply the IDE is more confused with booleans then I am.

Reply:
Doesn't matter, this was able to be written better :)